### DISCUSSION TITLE
Help dumping TG1 iinet Vant-5 v1.4 firmware for repository

### DISCUSSION THREAD

**Discussion by Limited-Testing** [hack-technicolor/hack-technicolor#222] [General]
Hi all,
 
I happen to have a TG-1 Vant-5 from iinet with the 15.32.1509-ver1.4 firmware which is missing from repository, per entry on this page.

https://hack-technicolor.readthedocs.io/en/stable/Repository/#tg-1-vant-5

If anyone wants to let me know how to dump the firmware, happy to have a go getting it off the router.

Asking first, as wasn't clear to me that current firmware can be retrieved, if I update firmware to different version.

![firmware version](https://user-images.githubusercontent.com/101632077/158347279-1c575e7f-e386-4091-9704-299e49f707e7.png)
.

**Comment by FrancYescO** [comment]
Root via any strategy, pretty sure #A will work for you https://hack-technicolor.readthedocs.io/en/latest/Hacking/Type2/

than follow standard procedure for dumps https://hack-technicolor.readthedocs.io/en/latest/Resources/#making-dumps

**Comment by LuKePicci** [comment]
Awesome let me know if you need help in getting such dumps!

**Comment by Limited-Testing** [comment]
Tried #A option AutoFlashGUI but didn't work to get root/ssh access.
Will samba option when have time.

**Comment by Limited-Testing** [comment]
Tried option #B FASTGate samba hack. But the url to set samba doesn't run.

ie
http://192.168.1.254/status.cgi?3g_pin=********&act=nvset&samba_enabled=1&samba_workgroup=WORKGROUP%5c%0a%09security%20%3d%20share%5c%0a%09guest%20account%20%3d%20root%5c%0a%09interfaces%20%3d%20lo%20br-lan%5c%0a%5c%0a%5bohnonotagain%5d&service=usb_status

Although because default ip is 10.1.1.1 used:
http://10.1.1.1/status.cgi?3g_pin=********&act=nvset&samba_enabled=1&samba_workgroup=WORKGROUP%5c%0a%09security%20%3d%20share%5c%0a%09guest%20account%20%3d%20root%5c%0a%09interfaces%20%3d%20lo%20br-lan%5c%0a%5c%0a%5bohnonotagain%5d&service=usb_status

Get page not found error.

**Reply by LuKePicci** [reply]
I suspect both #A and #C should work. I would retry #A with each of available "Generic" profiles in AFG

**Comment by Limited-Testing** [comment]
#C tch-exploit didn't work, it wasn't able to do the wps callback.

**Comment by Limited-Testing** [comment]
I didn't understand what the instructions are for #D external link goes to page 120 of a thread?

Tried #E and didn't work, seems not authenticating correctly.

~/firmware/FDEU-CVE-2020-1FC5$ python3 tg389ac_samba_exploit.py http://10.1.1.1 admin
Password:
[*] Init SRP authentication
[*] Get CSRF token
[*] Send authentication challenge
Traceback (most recent call last):
  File "tg389ac_samba_exploit.py", line 45, in <module>
    print ("[*] Error: start authentication, status code: " + res.status_code)
TypeError: can only concatenate str (not "int") to str

Any other ideas?

**Comment by Limited-Testing** [comment]
AutoFlashGUI  Generic Ping worked to get root ssh
(AutoFlashGUI  Generic Advanced DNS didn't work)

**Comment by LuKePicci** [comment]
Awesome, now go on with dumps as per instructions. I will give a look into your dumps in other to check whether #C strategy could also work or not (ofc that's a minor issue, #A is more then enough)

**Comment by Limited-Testing** [comment]
[dumps.zip](https://github.com/hack-technicolor/hack-technicolor/files/8357631/dumps.zip)

dumps of mtd3 and mtd4 in attached zip.

$cat /proc/mtd 
dev:    size   erasesize  name
mtd0: 08000000 00020000 "brcmnand.0"
mtd1: 02c60000 00020000 "rootfs"
mtd2: 01f80000 00020000 "rootfs_data"
mtd3: 02e60000 00020000 "bank_1"
mtd4: 02e60000 00020000 "bank_2"
mtd5: 00020000 00020000 "eripv2"
mtd6: 00040000 00020000 "rawstorage"
mtd7: 00000003 00020000 "blversion"

$ cat /proc/banktable/active
bank1
